### ISSUE TITLE
Syntax highlighting: Support for here-strings

### ISSUE THREAD

**Issue by askebv** [slynch8/10x#2174]
The Jai syntax isn't correct with comments inside here-strings (`#string`).
As far as I know, you can't solve this with regex alone. Because the way it works in Jai is that all text after `#string YourKeyword`
and up until
`\n` + 0..many whitespaces followed by `YourKeyword` is all auto-escaped by the compiler.

I would guess it needs a custom parser for here-strings.

![image](https://github.com/slynch8/10x/assets/4981659/8a887c58-d0e3-4ffa-9c71-b4c9c7a035cf)

**Comment by vulcan-dev** [comment]
I thought of this the other day, and figured it would be an unnecessary amount of work for it. I then thought of having an API function that can change the color of all the characters between x and y. I believe that is the best option, and seems easier?

**Comment by askebv** [comment]
> I thought of this the other day, and figured it would be an unnecessary amount of work for it. I then thought of having an API function that can change the color of all the characters between x and y. I believe that is the best option, and seems easier?

Alternatively it could also be a regex where group 2 is what needs to be colored:
`REGEXM(2, #string\s+(\w+)\s*\r?\n((?:.|\n)*?)\n(\1))`
defining it as `REGEXM(u8 match_group, char[] regex)`

It certainly works in the short-term

**Comment by vulcan-dev** [comment]
> > I thought of this the other day, and figured it would be an unnecessary amount of work for it. I then thought of having an API function that can change the color of all the characters between x and y. I believe that is the best option, and seems easier?
> 
> Alternatively it could also be a regex where group 2 is what needs to be colored: `REGEXM(2, #string\s+(\w+)\s*\r?\n((?:.|\n)*?)\n(\1))` defining it as `REGEXM(u8 match_group, char[] regex)`
> 
> It certainly works in the short-term

Yeah, that could work actually. I don't think there's a way to do regex like that now, but it would certainly work.
